src/RelayServer.ts

```diff
@@ -521,9 +538,8 @@ export class RelayServer extends EventEmitter {
             this.workerAddress
         );
 
-        const { feePercentage } = this.config;
-
-        if (feePercentage) {
+        if (!this.isSponsoredTx(req.relayRequest)) {
```

Comment:
```suggestion
        if (!this.isSponsoredTx(req.relayRequest) && feePercentage) {
```

Reply:
I take care of the of the undefined of the feePercentage here https://github.com/rsksmart/rif-relay-server/blob/PP-571/flexible-sponsorship/src/RelayServer.ts#L62.

By doing it, in that way, we will be only charging, if a feePercentage exists, and its not necessarily true.

Reply:
Yes, it makes sense indeed. 
The only "issue" is that right now we do add `feesPercentage` even if the value is undefined or 0. It's not a big issue, but just code that doesn't need to be executed.

Reply:
Im doing some testing right now, to avoid that :)

Reply:
I did some analysis and you are right, on line 541 we could add this conditional and avoid those lines to be executed, in  line 422 its a little more complicated. Therefore I was planning to leave as it is right now, to have uniformity, maybe we can have someone else input.

Reply:
Ok, no problem for me